rs/boundary_node/rate_limits/integration_tests/tests/rate_limit_canister_tests.rs

```diff
@@ -0,0 +1,247 @@
+    let result = pocket_ic
+        .update_call(
+            rate_limit_canister.into(),
+            sender.into(),
+            "add_config",
+            Encode!(&InputConfig {
+                schema_version: schema_version.into(),
+                rules,
+            })
+            .unwrap(),
+        )
+        .await
+        .map_err(|err| err.to_string())?;
+    let result = match result {
+        WasmResult::Reply(result) => result,
+        WasmResult::Reject(s) => panic!("Call to add_config failed: {:#?}", s),
+    };
+    Ok(Decode!(&result, AddConfigResponse).unwrap())
+}
+
+pub async fn get_config(
+    pocket_ic: &PocketIc,
+    rate_limit_canister: PrincipalId,
+    sender: PrincipalId,
+    version: Option<Version>,
+) -> Result<GetConfigResponse, String> {
+    let result = pocket_ic
+        .update_call( // could be query
+            rate_limit_canister.into(),
+            sender.into(),
+            "get_config",
+            Encode!(&version).unwrap(),
+        )
+        .await
+        .map_err(|err| err.to_string())?;
+    let result = match result {
+        WasmResult::Reply(result) => result,
+        WasmResult::Reject(s) => panic!("Call to get_config failed: {:#?}", s),
+    };
+    Ok(Decode!(&result, GetConfigResponse).unwrap())
+}
+
+#[tokio::test]
+async fn test() {
+    // 1. Prepare the world.
+    let pocket_ic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_ii_subnet()
+        .build_async()
+        .await;
+
+    let ii_subnet_id = {
+        let topology = pocket_ic.topology().await;
+        topology.get_ii().unwrap()
+    };
+
+    let with_mainnet_nns_canister_versions = false;
+    install_nns_canisters(
+        &pocket_ic,
+        with_mainnet_nns_canister_versions,
+        None,
+    )
+    .await;
+
+    let wasm = build_rate_limits_wasm();
+
+    let initial_payload = InitArg {
+        authorized_principal: None,
+        registry_polling_period_secs: 1,
+    };
+
+    let rate_limit_canister = install_canister(
+        &pocket_ic, 
+        "Rate Limits Canister", 
+        ii_subnet_id,
+        Encode!(&initial_payload).unwrap(), 
+        wasm.clone(),
+    ).await;
+
+    let wasm_hash = Sha256::hash(&wasm.clone().bytes());
+    assert_eq!(
+        get_installed_wasm_hash(&pocket_ic, rate_limit_canister).await,
+        wasm_hash,
+    );
+
+    // 2. Run code under test.
+
+    let add_config_response = add_config(
+        &pocket_ic,
+        rate_limit_canister,
+        PrincipalId::new_anonymous(),
+        0,
+        vec![],
+    ).await.unwrap();
+
+    println!("add_config_response = {:#?}", add_config_response);
+
+    let get_config_response = get_config(
+        &pocket_ic,
+        rate_limit_canister,
+        PrincipalId::new_anonymous(),
+        Some(0),
+    ).await.unwrap();
+
+    println!("get_config_response = {:#?}", get_config_response);
+
+    let new_wasm_bytes = modify_wasm_bytes(&wasm.clone().bytes(), 42);
```

Comment:
This inserts an arbitrary byte into the wasm, so the wasm hash is not the same as before. 

This would be redundant if we start from the mainnet WASM and upgrade to the tip-of-the-branch WASM in this test.